lib/metaheuristics/populationMetaheuristic.ts

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable operator-assignment, no-param-reassign */
+import EvolutionaryAlgorithm, { EvolutionaryConfig } from '../common/evolutionaryAlgorithm'
+import FitnessFunction from './fitnessFunction'
+import Random from '../math/random'
+
+export class PopulationMetaheuristicConfig extends EvolutionaryConfig {
+    public populationSize: number = 100
+}
+
+export class Individual {
+    public id: number = 0
+
+    public position: number[] = []
+
+    public fitness = Infinity
+
+    public bestPosition: number[] = []
+
+    public bestFitness = Infinity
```

Comment:
bestFitness should not be carried away by the individual - should be part of population